test/load-balancer-mocha.js

```diff
@@ -0,0 +1,90 @@
+            test.createNetwork({
+                balancer: "rdf-components/load-balancer"
+            }).then(function(network){
+                test.onOutPortData(network.processes.balancer.component, 'output', done);
+                fs.writeFile(targetFileName, "http://localhost:" + port + "/", function(){
+                    process.env['target-list-file'] = targetFileName;
+                    network.graph.addInitial('target-list-file', 'balancer', 'target_list_file_env');
+                    network.graph.addInitial("Hello", 'balancer', 'input');
+                });
+            }).catch(fail);
+        }).should.eventually.have.property('data', "Hello World!");
+    });
+    it("should distribute to multiple target", function() {
+        return new Promise(function(done, fail){
+            var server = http.createServer();
+            afterEach(_.once(server.close.bind(server)));
+            var count = 0;
+            var hash = {};
+            server.on('request', function(req, res) {
+                var body = [];
+                req.on('data', function(chunk) {
+                    body.push(chunk);
+                }).on('end', function() {
+                    body = Buffer.concat(body).toString();
+                    var json = JSON.parse(body);
+                    hash[req.url] = hash[req.url] || [];
+                    hash[req.url].push(json);
+                    res.end("Accepted");
+                    if (++count > 3) done(hash);
+                });
+            });
+            server.listen(port);
+            var targetFileName = path.join(os.tmpdir(), 'temp-target-list');
+            afterEach(_.once(fs.unlink.bind(fs, targetFileName)));
+            test.createNetwork({
+                balancer: "rdf-components/load-balancer"
+            }).then(function(network){
+                var list = [
+                    "http://localhost:" + port + "/one",
+                    "http://localhost:" + port + "/two",
+                    "http://localhost:" + port + "/three",
+                    "http://localhost:" + port + "/four",
+                    "http://localhost:" + port + "/five"
+                ];
+                fs.writeFile(targetFileName, list.join('\n'), function(){
+                    process.env['target-list-file'] = targetFileName;
+                    network.graph.addInitial('target-list-file', 'balancer', 'target_list_file_env');
+                    network.graph.addInitial({vnid:"A",data:"A"}, 'balancer', 'input');
+                    network.graph.addInitial({vnid:"B",data:"B"}, 'balancer', 'input');
+                    network.graph.addInitial({vnid:"C",data:"C"}, 'balancer', 'input');
+                    network.graph.addInitial({vnid:"D",data:"D"}, 'balancer', 'input');
+                });
+            }).catch(fail);
+        }).should.eventually.have.all.keys("/one", "/three", "/four", "/five");
+    });
+});
```

Comment:
These tests are good as far as they go.  Two more come to mind: 
1. A variation on your first test that floods several messages to a single target to verify that the messages are not dropped and we get response back.
2. Some error testing e.g., what happens if you don't have access the load balance is misconfigured e.g., no auth info, nothing to route to